### ISSUE TITLE
New rule: disallow-class

### ISSUE THREAD

**Issue by goloroden** [eslint/eslint#6254]
The `class` keyword is one of the most polarizing features of ES2015: You either love it or you hate it.

For the latter case it would be helpful to have a rule, `disallow-class`, that forbids usage of the `class` keyword, and hence forces you to stick to the classical, prototypic approach.

PS: There is a [bounty of 20$](https://www.bountysource.com/issues/34575790-new-rule-disallow-class) available for implementing this.

**Comment by ilyavolodin** [comment]
@goloroden You can use (no-restricted-syntax)[http://eslint.org/docs/rules/no-restricted-syntax] to get this functionallity. Configuration should be something like:

```js
"no-restricted-syntax": [2, "ClassDeclaration"]
```

This has been requested before, and we always said that we don't want to create separate rules for disabling part of JavaScript syntax (one exception is `no-with`), that's why we created `no-restricted-syntax` rule.
Closing as this is not something we are planning on doing.

**Comment by goloroden** [comment]
Perfect, thanks for pointing this out :-)
